CONTRIBUTION.md

```diff
@@ -0,0 +1,174 @@
+
+  1. Fork `apache/hadoop-ozone` github repository (first time)
+  2. Create a new Jira in HDDS project (eg. HDDS-1234)
+  3. Create a local branch for your contribution (eg. `git checkout -b HDDS-1234`)
+  4. Create your commits and push your branches to your personal fork.
+  5. Create a pull request on github UI 
+      * Please include the Jira link, problem description and testing instruction
+  6. Set the Jira to "Patch Available" state
+    
+## Code convention and tests
+
+We follow the code convention of Hadoop project (2 spaces instead of tabs, 80 char line width, ASF licence headers). The code checked with checkstyle, findbugs and various test frameworks.
+
+Please don't post / commit any code with any code violations (all checks are not checking the introduced violations as checks in Hadoop but all the available violations. One committed checkstyle issue will be reported in all of ht next pull requests.)
+
+### Check your contribution
+
+The easiest way to check your contribution is using the simplified shell scripts under `hadoop-ozone/dev-support/checks`. The problems will be printed out the standard output:
+
+For example:
+```
+hadoop-ozone/dev-support/checks/rat.sh
+hadoop-ozone/dev-support/checks/checkstyle.sh
+hadoop-ozone/dev-support/checks/findbugs.sh
+```
+
+Execution of rat and checkstyle are very fast. Findbugs is slightly slower. Executing unit.sh takes about 30 minutes.
+
+The same scripts are executed by the github PR checker.
+
+It's always good practice (and fast) to test with the related docker-compose based pseudo clusters:
+
+```
+cd hadoop-ozone/dist/target/ozone-*-SNAPSHOT/compose/ozone
+./test.sh
+```
+
+(To test S3 use `compose/ozones3`, to test security use `compose/ozonsecure`, etc.
+
+### False positive findbugs violation
+
+If you have __very good__ reasons, you can ignore any Fingbugs warning. Your good reason can be be persisted with the `@SuppressFBWarnings` annotation.
+
+```java
+@SuppressFBWarnings(value="AT_OPERATION_SEQUENCE_ON_CONCURRENT_ABSTRACTION",
+      justification="The method is synchronized and this is the only place "+
+          "dnsToUuidMap is modified")
+private synchronized void addEntryTodnsToUuidMap(
+...
+```
+
+## Using IDE
+
+As Ozone uses Apache Maven it can be developed from any IDE. As IntelliJ is a common choice here are some suggestions to use it for Ozone development:
+
+### Run Ozone from Intellij
+
+Ozone components depends on maven classpath. We generate classpath descriptor from the maven pom.xml files to use exactly the same classpath at runtime.
+
+As a result, it's easy to start _all_ the components from IDE as the right classpath (without provided scope) has already been set.
+
+To start Ozone from IntelliJ:
+
+1. Stop your IDE
+2. Execute the `./hadoop-ozone/dev-support/intellij/install-runconfigs.sh` helper script.
+3. Start the IDE
+4. New runner definitions are available from the Run menu.
+
+### Common problems
+
```

Comment:
Good idea, but let's do it in an other JIRA. I tried to add it, this is my draft:

```
### Setting up Checkstyle

Checkstyle plugin may help to detect violations directly from the IDE.

1. Install `Checkstyle+IDEA` plugin from `File` -> `Settings` -> `Plugins`
2. Open `File` -> `Settings` -> `Other settings` -> `Checkstyle` and Add (`+`) a new `Configuration File`
  * Description: `ozone`
  * Use a local checkstyle `./hadoop-hdds/dev-support/checkstyle/checkstyle.xml` 
3. Open the _Checkstyle Tool Window_, and select the ozone rule.
```

But it turned out that our checkstyle.xml is incompatible with IntelliJ. I can't import it as a rule because the location of the `LineLength` check. When I move the check to a different place, the command line checkstyle doesn't work.

It seems to be a bigger work to update.

Reply:
@anuengineer After reading your comment, I checked my local setup and sense this is definitely related to the checkstyle version. I am using IntelliJ's Checkstyle Plugin 5.24.2 which in turn uses Checkstyle 8.16 and I am still able to add the checkstyle file.

@elek What version do you have where it says invalid ? My IntelliJ says I have 4 pending updates to the plugin, I am on holding off on that :)

Reply:
you can update, but just choose a lower version in the combo box which says which version of CheckStyle needs to be used by the plugin interface in IntelliJ.

Reply:
> What version do you have where it says invalid ?

CheckStyle-IDEA 5.33.1

Reply:
>  I think if you choose up to check style version 8.6 or something the checkstyle.xml that is shipped with Hadoop will work in the IntelliJ IDE. 

Thanks @anuengineer , I missed this setting. I added the additional section about setting up checkstyle in IDE including this tip.